hasktorch/src/Torch/Functions.hs

```diff
@@ -242,3 +257,6 @@ all' t dim keepdim = unsafePerformIO $ (cast3 ATen.all_tlb) t dim keepdim
 
 any' :: Tensor -> Int -> Bool -> Tensor
 any' t dim keepdim = unsafePerformIO $ (cast3 ATen.any_tlb) t dim keepdim
+
+slice :: Tensor -> Int -> Int -> Int -> Int -> Tensor
```

Comment:
BTW We should come up with some kind of "named argument" approach to those ones, because this is severe `Int` blindness

Reply:
Totally ... there's a lot of Bool / Int / Tensor blindness at the libtorch level that's to be cleaned up in the API interfaces.

I did a little bit of work on this so far with little trivial sum types for Upper/Lower triangular matrices for solvers and reducers for the BCE loss. Trivially makes for nicer/less error-prone signatures while providing a mostly self-documenting interface as well.

For `slice` newtype Ints could be used to put names on the ordering and add a little safety.

Will keep cleaning this up as we curate the auto-generated interface..